backend/src/nodes/utils/utils.py

```diff
@@ -358,7 +358,7 @@ def convenient_upscale(
     else:
         # Add extra channels if not enough (i.e single channel img, three channel model)
         gray = False
-        if img.ndim == 2:
+        if input_channels > 1 and img.ndim == 2:
```

Comment:
Why did you add this?

Reply:
Ah, guess it's not necessary. Missed that it wouldn't duplicate dims if there was only one input channel.